src/reader.js

```diff
@@ -148,8 +148,12 @@ class Reader extends EventEmitter {
 
     // Trigger a query of the configured `lookupdHTTPAddresses`.
     const endpoint = this.roundrobinLookupd.next();
-    lookup(endpoint, this.topic, (err, nodes = []) =>
-      nodes.map(n => this.connectToNSQD(n.broadcast_address, n.tcp_port)));
+    lookup(endpoint, this.topic, (err, nodes = []) => {
+      if(err){
+        this.emit(Reader.ERROR, err);
+      }
```

Comment:
The first change makes sense! The lookupd based errors still doesn't make sense to me. We shouldn't kill the Reader just because an a lookupd service is having issues. If that's currently behavior then it's a bug and it needs to be fixed.